[dev-v2.6] nginx v1.2.0

kinarashah opened the pull request:
**Changelog**
----- 
- Using `v1.19.16-rancher1-5` and `v1.20.15-rancher1-3` from dev-v2.5 PR, to handle upgrade scenarios from Rancher v2.5.x to v2.6.x 

- Using  `nginx-v1.2.0-rancher1` which is upstream v1.2.0 version bump + template from dev-v2.5 PR. This is used by newly added following versions: 
    - `>=1.19.16-rancher1-5 <1.20.0-rancher1-1`
    - `>=1.20.15-rancher1-3 <1.21.0-rancher1-1` 
    - `>=1.21.12-rancher1-1  <1.22.0-rancher1-1` 
    
- Using `nginx-v1.2.0-rancher2` which is upstream v1.2.0 version bump + v2.6 changes, used by newly added following versions: 
    - `>=1.22.9-rancher1-1 <1.23.0-rancher1-1` 
    - `>=1.23.6-rancher1-1` 

 https://github.com/rancher/rancher/issues/37389

kinarashah committed:
add nginx v1.20

kinarashah committed:
go generate

snasovich commented on rke/k8s_rke_system_images.go:
```diff
@@ -5739,6 +5739,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			WindowsPodInfraContainer:  "rancher/kubelet-pause:v0.1.6",
 			Nodelocal:                 "rancher/mirrored-k8s-dns-node-cache:1.15.13",
 		},
+		// Enabled out of band post v2.5.12
```
Same as in the v2.5 PR we probably want it to reference v2.5.13.

snasovich commented on rke/k8s_rke_system_images.go:
```diff
@@ -6957,7 +7045,7 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			AciOpflexServerContainer:  "noiro/opflex-server:5.1.1.0.1ae238a",
 			PodInfraContainer:         "rancher/mirrored-pause:3.6",
-			Ingress:                   "rancher/nginx-ingress-controller:nginx-1.1.0-rancher1",
+			Ingress:                   "rancher/nginx-ingress-controller:nginx-1.2.0-rancher1",
```
For the line 7013 above we wanted to change comment from `// Enabled in v2.6.5` to `// Enabled out of band post v2.6.4`. Same is applicable to other versions introduced in https://github.com/rancher/kontainer-driver-metadata/pull/884.

snasovich reviewed (commented):
Couple of minor nits regarding comments on newly introduced/updated versions.

kinarashah commented:
- Updated comment to change from `Enabled out of band post v2.5.12` to `Enabled out of band post v2.5.13` & from `Enabled in v2.6.5` to `Enabled out of band post v2.6.4`. 

- Removed March patch updates (`v1.23.5-rancher1-1`, `v1.22.8-rancher1-1` and `v1.21.11-rancher1-1`)

snasovich reviewed (approved):
LGTM
